### ISSUE TITLE
When a user opts to create an account at checkout, error notices are missed on page reload

### ISSUE THREAD

**Issue by jjcoffee** [woocommerce/woocommerce-gateway-stripe#663]
Originally submitted as a WC issue here https://github.com/woocommerce/woocommerce/issues/20291
#### What I expected

On the checkout page if a user opts to create an account, and places an order the page reloads even if there are errors. 

On reload the browser should load at the top of the screen to display the error messages. Alternatively, the error messages could appear inline to the payment form.

#### What happened instead

On reload, browser automatically loads to the point where the user was last scrolled (which is usually the payment area). So if any error notices are present, they are likely to be missed by the user, causing confusion.

#### Steps to reproduce the issue

1. On checkout page, select to create an account
2. Scroll down the page
3. Enter some incorrect payment details
4. Place order, note the page will reload to the point it was scrolled to before, no error is visible

-------------------

- [ ] Issue assigned to next milestone.
- [ ] Issue assigned a priority (will be assessed by maintainers).

**Comment by roykho** [comment]
Can you please give more information on the testing steps? Such as did you actually fill in the create account name and email? And also what is the exact "incorrect payment details" did you use?

**Comment by jjcoffee** [comment]
Yes the whole form is filled correctly except for the payment area, which I entered one of Stripe's incorrect testing card numbers, e.g. address check fail (4000000000000010). 

The snippet I used to fix it for now is

`jQuery(window).on('load', function() {
	if(jQuery(".woocommerce-error").length>0){
		setTimeout( function(){
			jQuery("html, body").animate({ scrollTop: 0 }, "slow");
			return false;
		}, 1);
	}
});`

The setTimeout attempts to push it to the bottom of the call stack (I think), not sure how consistently it works, but it's been fine in testing so far...

**Comment by roykho** [comment]
Ok so after having tested this again, I am not so sure if the issue is with this plugin. In my test, I have commented out the scrolling to notice function and yet it still stays in the same place after refresh.

So another words, there is nothing that I am seeing from this plugin that is causing it to scroll there or stay there.

**Comment by jjcoffee** [comment]
Yeah it's just the standard browser behaviour to go to the last scrolled position on reload. But this causes confusion for users as they are unable to see error messages (and wouldn't necessarily think to scroll up). It's maybe more a feature request than a bug, I guess ;) Thanks for investigating though!
